Doc/glossary.rst

```diff
@@ -1103,10 +1103,10 @@ Glossary
 
       The :class:`collections.abc.Sequence` abstract base class
       defines a much richer interface that goes beyond just
-      :meth:`~object.__getitem__` and :meth:`~object.__len__`, adding
-      :meth:`count`, :meth:`index`, :meth:`~object.__contains__`, and
-      :meth:`~object.__reversed__`. Types that implement this expanded
-      interface can be registered explicitly using
+      :meth:`~object.__getitem__` and :meth:`~object.__len__`.
+      :ref:`Common Sequence Operations <typesseq-common>`
+      provides a list common to most sequence types.
+      Types which implement this expanded interface can be registered explicitly using
       :func:`~abc.ABCMeta.register`.
```

Comment:
```suggestion
      :meth:`~object.__getitem__` and :meth:`~object.__len__`, adding
      :meth:`!count`, :meth:`!index`, :meth:`~object.__contains__`, and
      :meth:`~object.__reversed__`. Types that implement this expanded
      interface can be registered explicitly using
      :func:`~abc.ABCMeta.register`. For more documentation on sesquence
      methods generally, see
      :ref:`Common Sequence Operations <typesseq-common>`.
```

Reply:
Hmmm... I know `__contains__` and `__reversed__` were there before, but what makes them special enough to call out here? `__getitem__` and `__len__` are the most basic sequence methods, so I understand why they are mentioned.

Reply:
The glossary entry is attempting here to distinguish between two definitions of "sequence" -- the loose concept of a sequence (`__getitem__` + `__len__`), and the more involved `collections.abc.Sequence` interface. I think it's useful for the entry to explain the difference between those two things, because they're easily confused. In order for the entry to explain the difference between those two things, it needs to list the extra methods that the `collections.abc.Sequence` interface adds on top of the "loose concept of sequence"

Reply:
> In order for the entry to explain the difference between those two things, **it needs to list the extra methods** that the collections.abc.Sequence interface adds on top of the "loose concept of sequence"

I guess this is what I don't get. Why enumerate these few methods and not all the others beyond `__getitem__` and `__len__`?

Reply:
> and not all the others beyond `__getitem__` and `__len__`?

Are there any others beyond `__getitem__` and `__len__` that `collections.abc.Sequence` adds and that aren't listed here? I can only see `__iter__`, which doesn't seem worth mentioning, since anything that implements `__getitem__` and `__len__` is actually already iterable even if it doesn't have an `__iter__` method. By contrast the fact that all `collections.abc.Sequence` instances have a `__contains__` method means that you can always use the `in` operator with them, and the fact that they all have a `__reversed__` method means that you can call `reversed()` on them